inference-engine/src/transformations/src/transformations/common_optimizations/remove_multi_subgraph_op_dangling_params.cpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/remove_multi_subgraph_op_dangling_params.hpp"
+#include "transformations/utils/utils.hpp"
+
+#include <memory>
+#include <vector>
+
+#include <openvino/opsets/opset8.hpp>
+#include <openvino/op/util/multi_subgraph_base.hpp>
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include <ngraph/rt_info.hpp>
+#include "itt.hpp"
+
+
+NGRAPH_RTTI_DEFINITION(ov::pass::RemoveMultiSubGraphOpDanglingParams, "RemoveMultiSubGraphOpDanglingParams", 0);
+
+ov::pass::RemoveMultiSubGraphOpDanglingParams::RemoveMultiSubGraphOpDanglingParams() {
```

Comment:
Also suggest to move into NopElimination

Reply:
@GlebKazantaev  Both transformations moved to NopElimination. As a result, I have to call NopElimination also during reshaping. For the model being fixed by this PR the exception caused by empty tensors is called during ConstantFolding (operations on HostTensor) so NopTransformation has to be called before ConstantFolding.

Reply:
@GlebKazantaev Unfortunately, after moving to NopElimnation (https://github.com/openvinotoolkit/openvino/pull/8035/commits/a2682d19d6208d2d1586c4c01639443789d9c80a) I have to change the order of NopElimination call (before CT). As a result, some other tests start to fail. 
I reverted it to state, where passes are split. Is it ok for you?

Reply:
Sorry that I probably confused you, but the idea was to have this new transformations as a separate transformation (as it was in your implementation) but instead of adding them explicitly to MOC pipeline I suggest to add them inside NopElimination pipeine. And for reshape method you still can use this transformations directly without calling NopElimination.